recipes/expat/all/test_package/CMakeLists.txt

```
@@ -1,10 +1,7 @@
 cmake_minimum_required(VERSION 3.1)
 project(test_package C)
 
-include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-conan_basic_setup(TARGETS)
-
 find_package(EXPAT REQUIRED)
 
 add_executable(${PROJECT_NAME} test_package.c)
-target_link_libraries(${PROJECT_NAME} EXPAT::EXPAT)
+target_link_libraries(${PROJECT_NAME} expat::expat)
```

```suggestion
target_link_libraries(${PROJECT_NAME} expat::expat)
```

Perfect example of misleading behavior due to https://github.com/conan-io/conan/issues/10387

---

Why do you think it is misleading? New `CMakeDeps` generators follows the value set at `cmake_target_name`, so it makes sense. Should we change it to `EXPAT::EXPAT` instead?

---

It's misleading because you call `find_package(EXPAT REQUIRED)`, so you should expect the target from https://cmake.org/cmake/help/latest/module/FindEXPAT.html. But since CMakeDeps forces `CMAKE_FIND_PACKAGE_PREFER_CONFIG` to ON, and it's not the default value, it breaks. This is why you have to use `expat::expat` instead of `EXPAT::EXPAT`.